ospf6d/ospf6_asbr.c

```diff
@@ -2153,25 +2156,65 @@ static const struct route_map_rule_cmd
 	ospf6_routemap_rule_set_metric_type_free,
 };
 
+struct ospf6_metric {
+	enum { metric_increment, metric_decrement, metric_absolute } type;
+	bool used;
+	uint32_t metric;
+};
+
 static enum route_map_cmd_result_t
 ospf6_routemap_rule_set_metric(void *rule, const struct prefix *prefix,
 			       void *object)
 {
-	char *metric = rule;
-	struct ospf6_route *route = object;
+	struct ospf6_metric *metric;
+	struct ospf6_route *route;
+
+	/* Fetch routemap's rule information. */
+	metric = rule;
+	route = object;
+
+	/* Set metric out value. */
+	if (!metric->used)
+		return RMAP_OKAY;
+
+	route->path.cost = route->path.redistribute_cost;
```

Comment:
This would be easier to read if you re moved this and handled the cases where it is applicable with

             route->path.cost = route->path.redistribute_cost + metric->metric; and
             route->path.cost = route->path.redistribute_cost - metric->metric;

Reply:
I changed that.